src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -3769,7 +3779,7 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
   }
 
   /** A class for type histories */
-  private final case class TypeHistory(var validFrom: Period, info: Type, prev: TypeHistory) {
+  private final case class TypeHistory(var validFrom: Period, var info: Type, var prev: TypeHistory) {
```

Comment:
Let's encapsulate these vars along the lines of https://github.com/retronym/scala/pull/73